### DISCUSSION TITLE
Server lobby UI overhaul feedback

### DISCUSSION THREAD

**Discussion by ghost** [FakeFishGames/Barotrauma#13728] [Unstable feedback]
### Change of server description to MOTD
Players can now only see the description as they join in text chat or look at a server in the serverlist. This adds space to the UI... but is bad for the following reasons:
- Servers tend to have rules which they place in the description, meaning players either have to read them as they join or be aware that the frankly miniscule server settings button has them. I've already experienced issues with players unaware of a server's rules since they don't read descriptions, I only see this making that issue even worse. The same applies for important info like unique game mechanics from mods.
- Servers often put their discord links in the server description, so that people can join it. People almost never join discord servers as they're joining, almost entirely doing so when _leaving_ a server, or said server is shutting down. If the description is more hidden and out of the way, people aren't as likely to be aware that such a server exists.
- If text chat is actively being used, then the MOTD will be pushed up and out of chat by people using chat as... chat quite quickly.

Also, the server settings button is too small. New hosts tend to have no idea that server settings exist already, and people who don't host will now rely on the button to read server MOTDs/descriptions outside of when they recently joined. If it's smaller and blends in more, people are going to notice it less and be generally less aware it exists.

**Comment by ven1883** [comment]
To add onto this,

- We don't need an entire tab for controlling respawn settings. That type of thing should stay relegated to the server settings menu, where it was previously.
- Placement of certain UI elements feels odd. I don't see the purpose of putting the game mode and settings tabs where they are. It feels like there is a lot of space taken up by these elements where they don't need it. It overall feels very... odd? Not haphazard, something a bit off.

**Reply by ghost** [reply]
> Placement of certain UI elements feels odd

It could be that you're used to a certain placement in all honesty. New changes often feel "odd" for that reason.

**Comment by Vykn** [comment]
I also think it's a bit odd that the respawn settings are so front and centre, taking up so much space on the server lobby, when they're a pretty small part of the game overall. It strikes me as more suitable to put them into the server settings or in the gamemode section

**Comment by SquidingTin** [comment]
Feels way to bright to me, the pure white text doesnt really flow with the rest being yellow or grey

**Comment by ven1883** [comment]
Given that this is out of unstable and now public, why?

This change absolutely cripples hosts and communities that actively use the description to serve rules and links to their community.
While you can still see the description, it's hidden deep, deep within the server settings which requires players to go out of their way to search for it rather than the convenient black box.

The MOTD system would be fine on its own, but the way it is handled is horrible. It uses the entire description box, which makes it completely unreadable.
The respawns tab is completely useless and should be relegated entirely to the Server Settings tab.

**Comment by ghost** [comment]
Recorded a video of accessing description via server settings.

https://github.com/FakeFishGames/Barotrauma/assets/86113712/777d9cdb-ed5a-413c-be43-6a1df7bd042f

I will say my opinions on this are mostly unchanged.

This is extremely hidden away, combined with the fact that *the server settings button was SHRINKED and recolored to BLEND IN in the same update* this means very few people will notice or read descriptions this way.
